lib/phoenix_live_view/html_formatter.ex

```diff
@@ -0,0 +1,424 @@
+    [
+      # ...omitted
+      heex_line_length: 300
+    ]
+    ```
+
+  ## Formatting
+
+  This formatter tries to be as consistent as possible with the Elixir formatter.
+
+  Given HTML like this:
+
+  ```eex
+    <section><h1>   <b>Hello</b></h1> </section>
+  ```
+
+  It will be formatted as:
+
+  ```eex
+  <section>
+    <h1><b><%= @user.name %></b></h1>
+  </section>
+  ```
+
+  A block element will go to the next line, while inline elements will be kept in the current line
+  as long as they fit within the configured line length.
+
+  The following links list all block and inline elements.
+
+  * https://developer.mozilla.org/en-US/docs/Web/HTML/Block-level_elements#elements
+  * https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements#list_of_inline_elements
+
+  It will also keep inline elements in their own lines if you intentionally write them this way:
+
+  ```eex
+  <section>
+    <h1>
+      <b><%= @user.name %></b>
+    </h1>
+  </section>
+  ```
+
+  This formatter will place all attributes on their own lines when they do not all fit in the
+  current line. Therefore this:
+
+  ```eex
+  <section id="user-section-id" class="sm:focus:block flex w-full p-3" phx-click="send-event">
+    <p>Hi</p>
+  </section>
+  ```
+
+  Will be formatted to:
+
+  ```eex
+  <section
+    id="user-section-id"
+    class="sm:focus:block flex w-full p-3"
+    phx-click="send-event"
+  >
+    <p>Hi</p>
+  </section>
+  ```
+
+  ### Intentional new lines
+
+  The formatter will keep intentional new lines. However, the formatter will
+  always keep a maximum of one line break in case you have multiple ones:
+
+  ```eex
+  <p>
+    text
+
+
+    text
+  </p>
+  ```
+
+  Will be formatted to:
+
+  ```
+  <p>
+    text
+
+    text
+  </p>
+  ```
+  """
+
+  alias Phoenix.LiveView.HTMLAlgebra
+  alias Phoenix.LiveView.HTMLTokenizer
+
+  # Default line length to be used in case nothing is specified in the `.formatter.exs` options.
+  @default_line_length 98
+
+  @behaviour Mix.Tasks.Format
+
+  @impl Mix.Tasks.Format
+  def features(_opts) do
+    [sigils: [:H], extensions: [".heex"]]
+  end
+
+  @impl Mix.Tasks.Format
+  def format(contents, opts) do
+    line_length = opts[:heex_line_length] || opts[:line_length] || @default_line_length
+
+    formatted =
+      contents
+      |> tokenize()
+      |> to_tree()
+      |> HTMLAlgebra.build(opts)
+      |> Inspect.Algebra.format(line_length)
+
+    # If the opening delimiter is a single character, such as ~H"...",
+    # do not add trailing newline.
+    newline = if match?(<<_>>, opts[:opening_delimiter]), do: [], else: ?\n
+
+    # TODO: Remove IO.iodata_to_binary/1 call on Elixir v1.14+
+    IO.iodata_to_binary([formatted, newline])
+  end
+
+  # Tokenize contents using EEx.tokenize and Phoenix.Live.HTMLTokenizer respectively.
+  #
+  # The following content:
+  #
+  # "<section>\n  <p><%= user.name ></p>\n  <%= if true do %> <p>deu bom</p><% else %><p> deu ruim </p><% end %>\n</section>\n"
```

Comment:
We should convert this text to english :D and make sure the tokens below are up to date!